receiver.py

```diff
@@ -0,0 +1,52 @@
+from time import sleep
+from typing import Union
+
+import daiquiri
+from celery import Celery
+from celery.events.state import Task, Worker
+
+from exporters import Exporter
+from utils import is_event_type_task
+
+logger = daiquiri.getLogger(__name__)
+
+
+class CeleryEventReceiver:
+    def __init__(self, broker: str):
+        self.broker = broker
+        self.celery_app = Celery(
+            broker=self.broker,
+        )
+        self.exporters: list[Exporter] = []
+        self.state = self.celery_app.events.State()
+
+    def attach(self, exporter):
+        self.exporters.append(exporter)
+
+    def notify(self, event: Union[Task, Worker]):
+        for exporter in self.exporters:
+            exporter.process_event(event)
+
+    def notify_event(self, event):
```

Comment:
Optional:
Related to above: This can be `.notify(event)` if the other one is renamed.